app/services/wizard_path_history.rb

```diff
@@ -22,6 +24,7 @@ def update
   end
 
   def previous_step
+    return path_history[PREVIOUS_STEP_INDEX] if step.nil?
```

Comment:
I feel like it would be clearer to specify the step in the controller. Is there a reason we shouldn't do that?

Reply:
Yes. Adding a step multiple times will also end up taking the user back to that step multiple times when pressing back, which is what I want to avoid.